content/en/docs/components/pipelines/sdk/output-viewer.md

```diff
@@ -225,6 +224,50 @@ Visualization of Scalar Metrics is as below:
   alt="V2 Scalar Metrics visualization"
   class="mt-3 mb-3 border border-info rounded">
 
+
+### Markdown
+
+Define `Output[Markdown]` parameter in your component function, then
+output Markdown content as string by writing to path like `<artifact_parameter_name>.path`. 
+Refer to
+[artifact_types.py](https://github.com/kubeflow/pipelines/blob/55a2fb5c20011b01945c9867ddff0d39e9db1964/sdk/python/kfp/v2/components/types/artifact_types.py#L420-L428) 
+for detail.
+
+```
+@component
+def markdown_visualization(markdown_artifact: Output[Markdown]):
+    markdown_content = '## Hello world \n\n Markdown content'
+    with open(markdown_artifact.path, 'w') as f:
+        f.write(markdown_content)
+```
+
+<img src="/docs/images/pipelines/v2/markdown-visualization.png" 
+  alt="Markdown visualization in v2 compatible mode"
+  class="mt-3 mb-3 border border-info rounded">
+
+
+
+### Single HTML file
+
+Define `Output[HTML]` parameter in your component function, then
+output HTML content as string by writing to path like `<artifact_parameter_name>.path`. 
+Refer to
+[artifact_types.py](https://github.com/kubeflow/pipelines/blob/55a2fb5c20011b01945c9867ddff0d39e9db1964/sdk/python/kfp/v2/components/types/artifact_types.py#L409-L417) 
+for detail.
+
+```
+@component
+def html_visualization(html_artifact: Output[HTML]):
+    html_content = '<!DOCTYPE html><html><body><h1>Hello world</h1></body></html>'
+    with open(html_artifact.path, 'w') as f:
+        f.write(html_content)
+```
+
+<img src="/docs/images/taxi-tip-analysis-step-output-webapp-popped-out.png" 
+  alt="Web app output from a pipeline component"
+  class="mt-3 mb-3 border border-info rounded">
```

Comment:
it's important to call out limitations of this HTML file (that it can only contain static refs to internet), you can refer to existing description for WebApp visualization.
https://github.com/kubeflow/website/blob/a4831ceb65c8d3755df686520e8875535bf00e2f/content/en/docs/components/pipelines/sdk/output-viewer.md#web-app

Reply:
Good catch! Updated the limitation about HTML file.